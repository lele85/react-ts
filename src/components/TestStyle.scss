@import '../styles/colors';

.componentBorder {
    border: 2px solid $accent_color;
}