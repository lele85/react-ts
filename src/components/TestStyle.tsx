import React from 'react';
import { StatelessComponent } from "react";
import styles from './TestStyle.scss';
import colors from '../styles/colors.scss';

export const TestStyle: StatelessComponent<{}> = () => {
    return (
        <div className={styles.componentBorder}>
            <h1>Colors from SCSS variables:</h1>
            <ul>
               <li style={{backgroundColor:colors.accentColor, color:"white"}}>{colors.accentColor}</li>
               <li style={{backgroundColor:colors.errorColor, color:"white"}}>{colors.errorColor}</li> 
               <li style={{backgroundColor:colors.warningColor, color:"white"}}>{colors.warningColor}</li> 
               <li style={{backgroundColor:colors.infoColor, color:"white"}}>{colors.infoColor}</li>  
            </ul>
        </div>
        
    );
}

export default TestStyle;