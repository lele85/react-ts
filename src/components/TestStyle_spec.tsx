import { shallow } from "enzyme";
import React from "react";
import TestStyle from "./TestStyle";

it("renders correctly", () => {
    const result = shallow(<TestStyle />);
    expect(result).toMatchSnapshot();
});
