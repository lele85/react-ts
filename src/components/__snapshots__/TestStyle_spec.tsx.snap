// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Colors from SCSS variables:
        </h1>,
        <ul>
          <li
            style={
              Object {
                "backgroundColor": "accentColor",
                "color": "white",
              }
            }
          >
            accentColor
          </li>
          <li
            style={
              Object {
                "backgroundColor": "errorColor",
                "color": "white",
              }
            }
          >
            errorColor
          </li>
          <li
            style={
              Object {
                "backgroundColor": "warningColor",
                "color": "white",
              }
            }
          >
            warningColor
          </li>
          <li
            style={
              Object {
                "backgroundColor": "infoColor",
                "color": "white",
              }
            }
          >
            infoColor
          </li>
        </ul>,
      ],
      "className": "componentBorder",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Colors from SCSS variables:",
        },
        "ref": null,
        "rendered": "Colors from SCSS variables:",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              style={
                Object {
                  "backgroundColor": "accentColor",
                  "color": "white",
                }
              }
            >
              accentColor
            </li>,
            <li
              style={
                Object {
                  "backgroundColor": "errorColor",
                  "color": "white",
                }
              }
            >
              errorColor
            </li>,
            <li
              style={
                Object {
                  "backgroundColor": "warningColor",
                  "color": "white",
                }
              }
            >
              warningColor
            </li>,
            <li
              style={
                Object {
                  "backgroundColor": "infoColor",
                  "color": "white",
                }
              }
            >
              infoColor
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "accentColor",
              "style": Object {
                "backgroundColor": "accentColor",
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "accentColor",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "errorColor",
              "style": Object {
                "backgroundColor": "errorColor",
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "errorColor",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "warningColor",
              "style": Object {
                "backgroundColor": "warningColor",
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "warningColor",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "infoColor",
              "style": Object {
                "backgroundColor": "infoColor",
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "infoColor",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Colors from SCSS variables:
          </h1>,
          <ul>
            <li
              style={
                Object {
                  "backgroundColor": "accentColor",
                  "color": "white",
                }
              }
            >
              accentColor
            </li>
            <li
              style={
                Object {
                  "backgroundColor": "errorColor",
                  "color": "white",
                }
              }
            >
              errorColor
            </li>
            <li
              style={
                Object {
                  "backgroundColor": "warningColor",
                  "color": "white",
                }
              }
            >
              warningColor
            </li>
            <li
              style={
                Object {
                  "backgroundColor": "infoColor",
                  "color": "white",
                }
              }
            >
              infoColor
            </li>
          </ul>,
        ],
        "className": "componentBorder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colors from SCSS variables:",
          },
          "ref": null,
          "rendered": "Colors from SCSS variables:",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                style={
                  Object {
                    "backgroundColor": "accentColor",
                    "color": "white",
                  }
                }
              >
                accentColor
              </li>,
              <li
                style={
                  Object {
                    "backgroundColor": "errorColor",
                    "color": "white",
                  }
                }
              >
                errorColor
              </li>,
              <li
                style={
                  Object {
                    "backgroundColor": "warningColor",
                    "color": "white",
                  }
                }
              >
                warningColor
              </li>,
              <li
                style={
                  Object {
                    "backgroundColor": "infoColor",
                    "color": "white",
                  }
                }
              >
                infoColor
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "accentColor",
                "style": Object {
                  "backgroundColor": "accentColor",
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": "accentColor",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "errorColor",
                "style": Object {
                  "backgroundColor": "errorColor",
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": "errorColor",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "warningColor",
                "style": Object {
                  "backgroundColor": "warningColor",
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": "warningColor",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "infoColor",
                "style": Object {
                  "backgroundColor": "infoColor",
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": "infoColor",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
